src/txmempool.cpp

```diff
@@ -205,7 +222,7 @@ bool CTxMemPool::CalculateMemPoolAncestors(const CTxMemPoolEntry &entry, setEntr
             if (setAncestors.count(parent_it) == 0) {
                 staged_ancestors.insert(parent);
             }
-            if (staged_ancestors.size() + setAncestors.size() + 1 > limitAncestorCount) {
+            if (staged_ancestors.size() + setAncestors.size() + total_count > limitAncestorCount) {
```

Comment:
I think staged ancestors is possibly overcounting here, need to run through if there are circumstances where we'd trigger early here.

Reply:
Yes, it's certainly overestimating for every case except for a package of parent + child. But I assume that's the vast majority of cases for a package.

I did just notice that we're double-counting the current tx though, so I'll update to do `staged_ancestors.size() + setAncestors.size() + total count - 1 > limitAncestorCount`